src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -59,7 +65,7 @@ internal ConnectorTypeMapper(NpgsqlConnector connector) : base(GlobalTypeMapper.
     {
         Connector = connector;
         UnrecognizedTypeHandler = new UnknownTypeHandler(Connector);
-        _resolvers = new List<TypeHandlerResolver>();
+        _resolvers = new TypeHandlerResolver[1];
```

Comment:
Any reason for the 1?

Reply:
The idea here was to support the case if someone calls `AddTypeResolverFactory` before `NpgsqlDatabaseInfo` is set. But since that shouldn't ever happen, changed to `Array.Empty` and added an assert.